CoreScriptsRoot/Modules/Panel3D.lua

```diff
@@ -250,40 +253,48 @@ function Panel3D.GetGUI(panel)
 end
 
 local zeroVector = Vector3.new(0, 0, 0)
-local baseHorizontal = CFrame.new()
-local basePosition = Vector3.new()
+local headHorizontal = CFrame.new()
+local headOffset = Vector3.new()
 local currentHoverPanel = nil
 local savedMouseBehavior = Enum.MouseBehavior.Default
 function Panel3D.OnRenderStep()
 	if not UserInputService.VREnabled then
 		return
 	end
+	local cameraCFrame = workspace.CurrentCamera.CFrame
 	local cameraRenderCFrame = workspace.CurrentCamera:GetRenderCFrame()
 	local userHeadCFrame = UserInputService:GetUserCFrame(Enum.UserCFrame.Head)
-	local cameraLook = cameraRenderCFrame.lookVector
+
+	local cameraLook = cameraCFrame.lookVector
 	local cameraHorizontalVector = Vector3.new(cameraLook.X, 0, cameraLook.Z).unit
-	local cameraPitchAngle = math.asin(cameraLook.Y)
+	local cameraHorizontal = CFrame.new(zeroVector, cameraHorizontalVector)
 
-	local position = workspace.CurrentCamera.CFrame.p
-	local panelsOrigin = CFrame.new(position) * baseHorizontal * CFrame.new(basePosition)
+	local userHeadLook = userHeadCFrame.lookVector
+	local userHeadHorizontalVector = Vector3.new(userHeadLook.X, 0, userHeadLook.Z).unit
+	local userHeadPitch = math.asin(userHeadLook.Y)
 
 	local isAboveThreshold = false
-	if cameraPitchAngle > panelLockThreshold or resetOrientationFlag then
+	if userHeadPitch > panelLockThreshold or resetOrientationFlag then
 		isAboveThreshold = true
-		baseHorizontal = CFrame.new(zeroVector, cameraHorizontalVector)
-		basePosition = userHeadCFrame.p
+		headHorizontal = CFrame.new(zeroVector, userHeadHorizontalVector)
+		headOffset = userHeadCFrame.p
 		resetOrientationFlag = false
 	end
 
+	local cameraPosition = workspace.CurrentCamera.CFrame.p
```

Comment:
We already have `cameraCFrame` above, we can increase performance by not having to re-fetch the camera data